tests/std/tests/P2093R14_formatted_output/test.cpp

```diff
@@ -0,0 +1,509 @@
+        FILE* file_stream_ptr;
+    };
+} // namespace test
+
+const locale& get_utf8_locale() {
+#pragma warning(push)
+#pragma warning(disable : 4640) // construction of local static object is not thread-safe
+    static const locale utf8_locale{"en-US.UTF-8"};
+#pragma warning(pop)
+
+    return utf8_locale;
+}
+
+wstring string_to_wstring(const string_view str) {
+    using facet_type = codecvt<wchar_t, char, mbstate_t>;
+    const facet_type& facet{use_facet<facet_type>(get_utf8_locale())};
+
+    mbstate_t conversion_state{};
+    const size_t num_chars_required = static_cast<size_t>(
+        facet.length(conversion_state, str.data(), str.data() + str.size(), (numeric_limits<size_t>::max)()));
+
+    wstring output_str;
+    output_str.resize_and_overwrite(num_chars_required, [&](wchar_t* const dest_str_ptr, const size_t output_size) {
+        const char* src_end_ptr;
+        wchar_t* dest_end_ptr;
+        const codecvt_base::result conversion_result = facet.in(conversion_state, str.data(), str.data() + str.size(),
+            src_end_ptr, dest_str_ptr, dest_str_ptr + output_size, dest_end_ptr);
+
+        assert(conversion_result == codecvt_base::ok);
+
+        return dest_end_ptr - dest_str_ptr;
+    });
+
+    return output_str;
+}
+
+string wstring_to_string(const wstring_view wide_str) {
+    using facet_type = codecvt<wchar_t, char, mbstate_t>;
+    const facet_type& facet{use_facet<facet_type>(get_utf8_locale())};
+
+    mbstate_t conversion_state{};
+    const size_t max_chars_required = wide_str.size() * facet.max_length();
+
+    string output_str;
+    output_str.resize_and_overwrite(max_chars_required, [&](char* const dest_str_ptr, const size_t output_size) {
+        const wchar_t* src_end_ptr;
+        char* dest_end_ptr;
+        const codecvt_base::result conversion_result = facet.out(conversion_state, wide_str.data(),
+            wide_str.data() + wide_str.size(), src_end_ptr, dest_str_ptr, dest_str_ptr + output_size, dest_end_ptr);
+
+        assert(conversion_result == codecvt_base::ok);
+
+        return dest_end_ptr - dest_str_ptr;
+    });
+
+    return output_str;
+}
+
+void maybe_flush_console_file_stream(const test::win_console& console) {
+    // std::print() and std::println() should automatically flush the stream if the Unicode
+    // API is being used, according to N4928 [print.fun]/7. So, as an additional check,
+    // we'll only call std::fflush() if the ordinary literal encoding is *NOT* UTF-8. This
+    // should work fine, assuming that our implementation is correct.
+    if constexpr (!_Is_ordinary_literal_encoding_utf8()) {
+        fflush(console.get_file_stream());
+    }
+}
+
+void initialize_console() {
+    // We want to do things like alter the output code page of our console, but we
+    // don't want to affect other tests which run concurrently. So, we want to detach this
+    // process from its current console, create a new console, and modify that one instead.

```

Comment:
This is spawning a bunch of console windows that rapidly flicker in and out of existence. Is there any way to avoid this?

Reply:
If there is no way to avoid this, perhaps we could make the "spawn windows" part run only in the PR/CI system by default, similar to how `/analyze` builds are PR/CI-only?

Reply:
It took me a whole day, 2 liters of Diet Mtn Dew, and half a large pizza, but I implemented exactly that - creating a new process with a new console and a hidden window, then using Windows semaphores for inter-process communication. We need two semaphores - one for the parent process to wait for the child process to start up (otherwise we can't attach to its console instantaneously), and another for the child process to wait for the parent process to finish its testing (otherwise, if the child process tries to exit too early, the parent process will fail). We don't need to communicate test status, since the originally-started parent process is the one that runs the tests and returns an exit code to the test infrastructure, it just scribbles into the child process's hidden console. After developing this in an example program with lots of prints and sleeps, it seems to be quite robust, and that transferred over to the multithreaded test harness - I observe zero flickering and reliable passing.